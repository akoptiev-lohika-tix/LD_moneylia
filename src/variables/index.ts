export * from './text-variables';
export * from './style-variables';