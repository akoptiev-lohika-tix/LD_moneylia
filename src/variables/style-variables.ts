export const colors = {
  common: {
    white: '#ffffff',
  },
  primary: {
    main: '#3a3b7b',
    light: '#f8f4ff',
    dark: '#7476ed',
  },
  grey: {
    main: '#6f6d7b',
    dark: '#312e43',
    light: '#83828e',
  },

  error: {
    main: '#f83e5a',
  },
};

export const shadows = {
  shadow1: {
    shadowColor: 'rgba(0, 0, 0, 0.08)',
    shadowOffset: { width: 0, height: 0 },
  },
};
