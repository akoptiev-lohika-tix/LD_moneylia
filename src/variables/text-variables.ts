// General
export const ERROR_GENERAL_MESSAGE =
  'Oops...Your request is unavailable now. Please try again later or contact us';

// Join-screen
export const JOIN_SCREEN_TITLE = 'Save your money';
export const JOIN_SCREEN_DESCRIPTION =
  'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Est in quis erat a sit.';
export const JOIN_SCREEN_BUTTON_TEXT = 'Join for free';
export const JOIN_SCREEN_LINK_TEXT = "Don't have SPID or CIE? Find out more";

// Dashboard
export const WELCOME_TEXT = 'Welcome in Moneylia';
export const SUMMARY_TITLE = 'User summary';
export const USER_SUMMARY_STATUS_TITLE = 'Status';
export const USER_SUMMARY_BIRTHDAY_TITLE = 'Birthday';
export const USER_SUMMARY_CITY_TITLE = 'City';
export const USER_SUMMARY_COUNTRY_TITLE = 'Country';
export const USER_SUMMARY_NUMBER_TITLE = 'Number';
export const DASHBOARD_CONTRIBUTIONS_PAID_TITLE = 'Paid';
export const DASHBOARD_CONTRIBUTIONS_DUE_TITLE = 'Due';
export const DASHBOARD_CONTRIBUTIONS_ADVANCED_PAYMENT_TITLE = 'Advance payment';
export const DASHBOARD_CONTRIBUTIONS_MODULARITY_TITLE = 'Modularity';

// Not Found Screen
export const NOT_FOUND_SCREEN_TEXT =
  'Seems like this part of app is developing now. Please come back later';

// Tax Payments
export const TAX_PAYMENTS_TOP_BAR_TITLE = 'Tax payments';
export const TAX_PAYMENTS_CONTENT_TITLE = 'Movements to pay';
export const TAX_PAYMENTS_CONTENT_DESCRIPTION =
  'View pending transactions for the past two years and proceed to payment.';
