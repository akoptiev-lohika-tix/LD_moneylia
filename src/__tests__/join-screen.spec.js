import React from 'react';
import { render, screen, waitFor } from '@testing-library/react-native';
import renderer from 'react-test-renderer';

import JoinScreen from '../screens/join-screen/join-screen';
import { JOIN_SCREEN_TITLE } from '../variables';

test('JoinScreen renders correctly', async () => {
  const tree = await waitFor(() => renderer.create(<JoinScreen />).toJSON());
  expect(tree).toMatchSnapshot();
});

test('should have correct title', async () => {
  await waitFor(() => render(<JoinScreen />));

  expect(screen.findByText(JOIN_SCREEN_TITLE)).toBeTruthy();
});

test('should have button', async () => {
  await waitFor(() => render(<JoinScreen />));

  expect(screen.findByTestId('join-button')).toBeTruthy();
});
