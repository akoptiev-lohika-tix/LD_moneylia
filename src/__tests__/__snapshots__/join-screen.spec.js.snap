// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JoinScreen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "height": "100%",
      "paddingHorizontal": 20,
      "paddingTop": 90,
    }
  }
>
  <View
    style={
      Object {
        "marginBottom": 24,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/join-screen-image.png",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "paddingHorizontal": 24,
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#3a3b7b",
            "fontFamily": "Work Sans SemiBold",
            "fontSize": 32,
            "lineHeight": 38,
          }
        }
      >
        Save your money
      </Text>
    </View>
    <View
      style={
        Object {
          "marginBottom": 60,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#6f6d7b",
            "fontFamily": "Work Sans Regular",
            "fontSize": 14,
            "letterSpacing": -0.5,
            "lineHeight": 21,
            "textAlign": "center",
          }
        }
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Est in quis erat a sit.
      </Text>
    </View>
    <View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "#7476ed",
              "borderRadius": 4,
              "marginBottom": 24,
              "paddingHorizontal": 75,
              "paddingVertical": 16,
            },
            null,
          ]
        }
        testID="join-button"
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontFamily": "Work Sans SemiBold",
              "fontSize": 18,
              "lineHeight": 18,
            }
          }
        >
          Join for free
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
          },
          null,
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#3a3b7b",
            "fontFamily": "Rubik Regular",
            "fontSize": 14,
            "lineHeight": 21,
            "textDecorationLine": "underline",
          }
        }
      >
        Don't have SPID or CIE? Find out more
      </Text>
    </View>
  </View>
</View>
`;
